### PR TITLE
BSP-128: Form A section 1 further details of app

### PR DESCRIPTION
**Pull request by johnoneill46**
Added validation and transformation for the following fields inside Form A section 1:
- 'AddressofProperties'
- 'MortgageDetails'
- 'OrderForChildren'
- 'OrderForChildrenNoAgreement'
- 'ChildSupportAgencyCalculationMade'
- 'ChildSupportAgencyCalculationReason'

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java [hunk 1]
```diff
@@ -65,23 +72,43 @@
                 "ApplyToVary",
                 "ApplyToDischargePeriodicalPaymentOrder"
         ));
-        ALLOWED_VALUES_PER_FIELD.put(APPLYING_FOR_CONSENT_ORDER, asList("Yes"));
+        ALLOWED_VALUES_PER_FIELD.put(APPLYING_FOR_CONSENT_ORDER, Collections.singletonList("Yes"));
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java]: so many bytes wasted!

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java [hunk 2]
```diff
@@ -65,23 +72,43 @@
...
-    
+
     @Override
     protected Map<String, List<String>> getAllowedValuesPerField() {
         return ALLOWED_VALUES_PER_FIELD;
     }
 
     @Override
     protected List<String> runPostProcessingValidation(Map<String, String> fieldsMap) {
-        List<String> errorMessages = Stream.of(
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java]: I don't know, I actually like it says `errorMessages` as it tells me what this thing returns

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java [hunk 3]
```diff
@@ -141,7 +166,7 @@
     private static List<String> validateHwfNumber(Map<String, String> fieldsMap, String fieldName) {
         String hwfNumber = fieldsMap.get(fieldName);
         return hwfNumber != null && !hwfNumber.matches(HWF_NUMBER_6_DIGITS_REGEX)
-            ? asList("HWFNumber is usually 6 digits")
+            ? Collections.singletonList("HWFNumber is usually 6 digits")
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java]: ditto - is it better?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java (hunk 1 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java]: 🤷‍♂ IIRC it's actually identical amount of bytes used, just Collections.singletonList is a superrrr small amount more performant -  I'll revert it though sure

Comment (jakub-kozlowski) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/validation/FormAValidator.java]: :-D I meant source code bytes - thank you though - it's about source code reading comfort

**Pull request merged**